x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/components/deduplication_panel/deduplication_panel.tsx

```diff
@@ -0,0 +1,219 @@
+            defaultMessage="The web crawler only indexes unique pages. Choose which fields the crawler should use when
+            considering which pages are duplicates. Deselect all schema fields to allow duplicate
+            documents on this domain. {documentationLink}."
+            values={{
+              documentationLink: (
+                <EuiLink
+                  href={`${DOCS_PREFIX}/web-crawler-reference.html#web-crawler-reference-content-deduplication`}
+                  target="_blank"
+                  external
+                >
+                  {i18n.translate(
+                    'xpack.enterpriseSearch.appSearch.crawler.deduplicationPanel.learnMoreMessage',
+                    {
+                      defaultMessage: 'Learn more about content hashing',
+                    }
+                  )}
+                </EuiLink>
+              ),
+            }}
+          />
+        </p>
+      </EuiText>
+      <EuiSpacer size="l" />
+      <EuiSwitch
+        label="Prevent duplicate documents"
+        checked={deduplicationEnabled}
+        onChange={() =>
+          deduplicationEnabled
+            ? submitDeduplicationUpdate(domain, { enabled: false, fields: [] })
+            : submitDeduplicationUpdate(domain, { enabled: true })
+        }
+      />
+      <EuiFlexGroup>
+        <EuiFlexItem>
+          <div className="selectableWrapper">
+            <EuiSelectable
+              options={selectableOptions}
+              onChange={(options) =>
+                submitDeduplicationUpdate(domain, {
+                  fields: getCheckedOptionLabels(options as Array<EuiSelectableLIOption<object>>),
+                })
+              }
+              searchable
+              searchProps={{
+                disabled: !deduplicationEnabled,
+                append: (
+                  <EuiPopover
+                    button={
+                      <EuiButtonEmpty
+                        size="xs"
+                        iconType="arrowDown"
+                        iconSide="right"
+                        onClick={() => setShowAllFieldsPopover(!showAllFieldsPopover)}
+                        className="showAllFieldsPopoverToggle"
+                        disabled={!deduplicationEnabled}
+                      >
+                        {showAllFields
+                          ? i18n.translate(
+                              'xpack.enterpriseSearch.appSearch.crawler.deduplicationPanel.allFieldsLabel',
+                              {
+                                defaultMessage: 'All fields',
+                              }
+                            )
+                          : i18n.translate(
+                              'xpack.enterpriseSearch.appSearch.crawler.deduplicationPanel.selectedFieldsLabel',
+                              {
+                                defaultMessage: 'Selected fields',
+                              }
+                            )}
```

Comment:
Not sure what the preferred code style is in Kibana but I would perhaps extract this to a `const` and use that value here.

Reply:
We're not super consistent about this, for the most part I prefer inline if its not being re-used elsewhere.